### ISSUE TITLE
Laravel Passport keys not available in Lambda

### ISSUE THREAD

**Issue by Rezrazi** [brefphp/bref#1667]
**Description:**

Having Passport installed, and following the instructions described here https://bref.sh/docs/laravel/passport
I'm unable to get Passport working, throwing an exception: `Invalid key supplied`

Investigation so far:
- serverless.yml is updated to include oauth keys into the bundle
- Downloading the zip manually from S3 does indeed show the keys being included in the bundle (cf. screenshot)
- Dumping the contents from inside the Lambda 

```php
        $tmp = Storage::build('/tmp');

        dd(
            $tmp->allFiles(),
            $tmp->allDirectories(),
            storage_path(),
            shell_exec(
                'ls -la ' . storage_path()
            )
        );
```
![CleanShot 2023-10-17 at 22 00 41@2x](https://github.com/brefphp/bref/assets/2086576/4b2603fd-621f-45de-8ce9-f0a2eea05bc0)

![CleanShot 2023-10-17 at 21 54 16@2x](https://github.com/brefphp/bref/assets/2086576/2f1e3e0b-5b7b-46f8-83a1-4a504076fc35)

**How to reproduce:**

- Install Laravel
- Install laravel/passport
- Generate passport keys and persist them in serverless.yml
- Deploy

**Comment by GrahamCampbell** [comment]
Probably you should not try to store the private key in plain text in the lambda image, but instead load it via secrets manager into an env variable at runtime.

**Comment by mnapoli** [comment]
Note that this is indeed a regression with the Bref bridge v2.

It's because the BrefServiceProvider sets `storage_path` to `/tmp/storage` now. We may want to copy any file in `storage/` being deployed into `/tmp/storage` (that's what we do in the Symfony bridge for example). Or maybe there's a better alternative to imagine.

**Comment by Rezrazi** [comment]
@GrahamCampbell Yeah, I did end up doing that. I was experimenting with an existing app when I noticed this behavior

@mnapoli sorry if I'm not too knowledgeable with Bref v1, but was there any particular reason to put internals in a /tmp/storage path?

**Comment by mnapoli** [comment]
@Rezrazi anything outside `/tmp` is read-only, so it was creating problems when using the Laravel cache.

**Comment by wojo1206** [comment]
> Probably you should not try to store the private key in plain text in the lambda image, but instead load it via secrets manager into an env variable at runtime.

Please remember that lambda `ENV` has 4K limit! RSA keypair generated by `php artisan passport:keys` could easily meet the limit.

**Comment by wojo1206** [comment]
I followed the documentation for Laravel Passport setup. I think the documentation isn't clear or the solution for this still work in progress. My setup as (as described in docs):  

```yml
# serverless.yml

package:
    patterns:
        - ...
        # Exclude the 'storage' directory
        - '!storage/**'
        # Except the public and private keys required by Laravel Passport
        - 'storage/oauth-private.key'
        - 'storage/oauth-public.key' 
```

On `serverless deploy` they keys are store at `storage/` path not at `/tmp/storage` as one might expect! Luckily, Passport can adjust the path with `Passport::loadKeysFrom('storage');`

**Comment by mnapoli** [comment]
Thanks for sharing the workaround! If you have the time for a pull request (https://github.com/brefphp/bref/blob/master/docs/laravel/passport.mdx) that would be awesome!

**Comment by wojo1206** [comment]
I don't know if the behavior I observed is expected. Maybe on deploy `bref` could copy contents of package (as defined in serverless.yml) into `/tmp/storage` and never use `storage`? Right now, `bref` maintains two `storage` paths.

**Comment by mnapoli** [comment]
Ideally it should copy the files, yes. But that's not implemented right now, so in the meantime mentioning the trick in the documentation is the best option I think.

**Comment by maulikpatelbtech** [comment]
@mnapoli Please let me know if you've discovered a method to copy any file located in `storage/` to be deployed into `/tmp/storage`.

> Note that this is indeed a regression with the Bref bridge v2.
> 
> It's because the BrefServiceProvider sets `storage_path` to `/tmp/storage` now. We may want to copy any file in `storage/` being deployed into `/tmp/storage` (that's what we do in the Symfony bridge for example). Or maybe there's a better alternative to imagine.

**Comment by cirolosapio** [comment]
On **Laravel 13 + Passport 13**, the workaround needs two lines in `AppServiceProvider::boot()`.

First, note that the relative path `Passport::loadKeysFrom('storage')` mentioned in the bref docs may have worked in the past, but on Laravel 13 I had to use `base_path('storage')` explicitly, otherwise the keys are still not found.

Second, Passport 13 introduced strict key permission validation (laravel/passport#1789), and since Lambda mounts the ZIP as read-only with `755` permissions, the check always fails.

The full fix:

```php
// Keys are deployed in the ZIP at /var/task/storage (not /tmp/storage where storage_path() points on Lambda)
Passport::loadKeysFrom(base_path('storage'));
// Lambda mounts ZIP files as read-only with 755 permissions; skip the check
Passport::$validateKeyPermissions = false;
```
